Anagram.java

```diff
@@ -29,22 +28,63 @@ public static void main(String args[]) {
 
 	// Returns true if the two given strings are anagrams, false otherwise.
 	public static boolean isAnagram(String str1, String str2) {
-		// Replace the following statement with your code
-		return false;
+		str1=preProcess(str1);
+		str2=preProcess(str2);
+		boolean check = true;
+		String str2temp = str2;
+		for (int i=0;i<str1.length();i++){
+			check = false;
+			str2= str2temp; 
+			for (int j=0;j<str2.length();j++) {
+				str2temp = "";
+				if (str1.charAt(i) == str2.charAt(j)) {
+					check = true;
+					for (int t=0;t<str2.length();t++){
+						if (t!=j) {
+							str2temp=str2temp + str2.charAt(t); 
+						}
+					}
+					j=str2.length();
```

Comment:
1. Bad names .. 
2. Dont use more than 4 scopes in a function. in this course, if you reach a case where you have more than 4 scopes it means that 
- If you have exported some into an independent helper function (see recitation 3, isValidPassword question)
- you can build some of your conditions better into differently.

Please come to my office hour on Friday. I want to help you rebuild this function..